src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java

```diff
@@ -44,7 +44,15 @@ public LostItemFeeRefundService(Clients clients) {
     CheckInContext checkInContext) {
 
     return refundLostItemFees(forCheckIn(checkInContext))
-      .thenApply(r -> r.map(context -> checkInContext.withLoan(context.getLoan())));
+      .thenApply(r -> r.map(context -> {
+        // check-in of item without an open loan
+        // for such check-in we should not set loan in response.
+        if (checkInContext.getLoan() == null) {
+          return checkInContext;
+        }
+
+        return checkInContext.withLoan(context.getLoan());
```

Comment:
How does this ensure that the loan is not open?

Reply:
Actually, we don't need to ensure that loan is still open. When we check in a `Lost and paid` item, there will be no loan associated to the item, therefor `checkInContext.getLoan()` will return a `null`, but the refund logic is flexible and it attempts to find a closed loan for the `Lost and paid` item in order to refund lost fees. Because of that logic, there will be a loan in RefundContext, but we should not set it to the check in context, because the closed loan is not affected by the current check in operation.

Reply:
Maybe it's worth changing the comment to explain that (see suggestion).

Is the loan not fetched elsewhere or only during refunds? If it is, could this copy be skipped entirely?

```suggestion
        // Refunds may operate on open or closed loans, 
        // closed loans should not be included in responses for check ins
        // so the loan should not be copied to the check in context
        if (checkInContext.getLoan() == null) {
          return checkInContext;
        }

        return checkInContext.withLoan(context.getLoan());
```

Reply:
We fetch and set accounts for the loan and lost item fee policy during refund, so that data is used in overdue fine service and it uses these records to decide whether to charge a fine or not. Setting the updated loan with accounts and lost item policy is important here.

Reply:
Does that mean that the loan does get changed during the refund process?

Reply:
Not the loan representation, but Loan circulation object (reference records that are injected into the loan).